docs/mp/persistence.adoc

```diff
@@ -0,0 +1,1407 @@
+  `DataSource` knows how to connect to it
+
+* Information required for the vendor-supplied `DataSource` to
+  authenticate to the database and otherwise tailor itself to it
+
+Some examples for representative configurations follow.  The list of
+such configurations is not exhaustive.
+
+==== Configuration Prefixes [[DS-Configuration-Prefixes]]
+
+All MicroProfile Config-compatible property names for Helidon MP's
+named data source integration follow a common pattern:
+
+[example]
+*_objecttype_*.*_datasourcename_*.*_propertyname_*
+
+* The name of a given configuration property always begins with the
+  *_objecttype_* portion: a fully-qualified Java class name of the
+  object being configured.  Configuration for Helidon MP's named
+  data source integration concerns the behavior of
+  `javax.sql.DataSource` objects, so Helidon MP's named data source
+  integration configuration property names begin with
+  `javax.sql.DataSource`.
+
+** A period (`.`) separates the _objecttype_ portion from the rest of
+   the property name.
+
+* The *_datasourcename_* portion, the name of the data source being
+  configured, comes next.  It cannot contain a period (`.`).
+
+** A period (`.`) separates the _datasourcename_ portion from the rest
+   of the property name.
+
+* The *_propertyname_* portion, identifying the connection-pool- or
+  vendor-supplied-`DataSource`-specific configuration property name,
+  comes last.  It may contain periods (`.`).
+
+As an example, configuration to set an imaginary `foo.bar` property on
+the `test` data source's associated connection pool or vendor-specific
+`DataSource` to `baz` looks like this in Java `.properties` format:
+
+[source,properties]
+----
+javax.sql.DataSource.test.foo.bar=baz # <1> <2> <3>
+----
+<1> The *_objecttype_* portion of the configuration property name is
+    `javax.sql.DataSource`.
+<2> The *_datasourcename_* portion of the configuration property name
+    is `test`.
+<3> The *_propertyname_* portion of the configuration property name is
+    `foo.bar`.
+
+==== Examples [[DS-Configuration-Examples]]
+
+Here are some examples illustrating general named data source
+configuration patterns in various
+xref:config/advanced-configuration.adoc[common MicroProfile
+Config-compatible locations].
+
+===== Example: `META-INF/microprofile-config.properties` Classpath Resource
+
+Here is an example of some named data source configuration as might be
+found in a
+`src/main/resources/META-INF/microprofile-config.properties`
+configuration source:
+
+[source,properties]
+----
+javax.sql.DataSource.yourDataSourceName.somePropertyOfYourConnectionPoolAndDataSource = itsValue
+javax.sql.DataSource.yourDataSourceName.someOtherPropertyOfYourConnectionPoolAndDataSource = anotherValue
+----
+
+===== Example: System Properties Set on the Command Line
+
+Here is an example of some named data source configuration using
+system properties on the command line instead:
+
+[source,shell]
```

Comment:
There is currently no other document using `[source,shell]` ; they all use `[source,bash]`.
The language attributes is used as a CSS class for the syntax highlighting. This means potential differences in the colors used for the syntax highlighting when compared to other similar snippets.

Reply:
Fixed